packages/twenty-front/src/modules/workflow/hooks/useRightDrawerWorkflowSelectAction.tsx

```diff
@@ -0,0 +1,129 @@
+import { useTabList } from '@/ui/layout/tab/hooks/useTabList';
+import { useCreateNode } from '@/workflow/hooks/useCreateNode';
+import { createStepFromParentStepIdState } from '@/workflow/states/createStepFromParentStepIdState';
+import { showPageWorkflowDiagramTriggerNodeSelectionState } from '@/workflow/states/showPageWorkflowDiagramTriggerNodeSelectionState';
+import { Workflow } from '@/workflow/types/Workflow';
+import { useRecoilValue, useSetRecoilState } from 'recoil';
+import {
+  IconPlaystationSquare,
+  IconPlug,
+  IconPlus,
+  IconSearch,
+  IconSettingsAutomation,
+} from 'twenty-ui';
+import { v4 } from 'uuid';
+import { logError } from '~/utils/logError';
+
+export const useRightDrawerWorkflowSelectAction = ({
+  tabListId,
+  workflow,
+}: {
+  tabListId: string;
+  workflow: Workflow;
+}) => {
+  const createStepFromParentStepId = useRecoilValue(
+    createStepFromParentStepIdState,
+  );
+
+  const setShowPageWorkflowDiagramTriggerNodeSelection = useSetRecoilState(
+    showPageWorkflowDiagramTriggerNodeSelectionState,
+  );
+
+  const { createNode } = useCreateNode({ workflow });
+
+  const allOptions: Array<{
+    id: string;
+    name: string;
+    type: 'standard' | 'custom';
+    icon: any;
+  }> = [
+    {
+      id: 'create-record',
+      name: 'Create Record',
+      type: 'standard',
+      icon: IconPlus,
+    },
+    {
+      id: 'find-records',
+      name: 'Find Records',
+      type: 'standard',
+      icon: IconSearch,
+    },
+  ];
+
+  const tabs = [
+    {
+      id: 'all',
+      title: 'All',
+      Icon: IconSettingsAutomation,
+    },
+    {
+      id: 'standard',
+      title: 'Standard',
+      Icon: IconPlaystationSquare,
+    },
+    {
+      id: 'custom',
+      title: 'Custom',
+      Icon: IconPlug,
+    },
+  ];
+
+  const { activeTabIdState } = useTabList(tabListId);
+  const activeTabId = useRecoilValue(activeTabIdState);
+
+  const options = allOptions.filter(
+    (option) => activeTabId === 'all' || option.type === activeTabId,
+  );
+
+  const handleActionClick = async (actionId: string) => {
+    try {
```

Comment:
do we need a try catch here? It's very rare we use it in the FE only when we use underlying libraries throwin exceptions